taurus/client/json_encoder.py

```diff
@@ -153,7 +153,7 @@ def thin_dumps(obj, **kwargs):
     """
     set_uuids(obj)
     res = deepcopy(obj)
```

Comment:
Can we get rid of this deepcopy now that `substitute_links` does not mutate?

Reply:
probably, I'll look around for `deepcopy`s to remove